### PR TITLE
Move BatchJob to Separate Module

### PR DESCRIPTION
**Pull request by dthain**
Note that the dependency tree is now:
```
makeflow wavefront: batch_job
batch_job: work_queue chirp
```
`work_queue_pool` seems like it's in the wrong place at first, but it only invokes `batch_job` not `libwork_queue` so it makes sense to stay in `batch_job`

This is a preparatory step to clean up our makefiles...

### CODE REVIEW

**Comment by dthain** [comment]
@nhazekam @btovar how does this look?

**Comment by nhazekam** [comment]
The move looks good to me. I understand why we are moving work_queue_pool, but do we also want to move *_submit_workers? They seem like it could go either.

**Comment by dthain** [comment]
Fixed commits lost in the merge.

Well, the idea is that work queue can stand alone, separately of the batch_job module.  If you wanted to build and install work_queue by itself, then you would also want the submit scripts, which don't use the batch job module.

**Comment by nhazekam** [comment]
OK, makes sense to me.

File: batch_job/src/Makefile~ [hunk 1]
```diff
@@ -0,0 +1,38 @@
+include ../../config.mk
```
Comment (btovar) [inline batch_job/src/Makefile~]: This is a backup file (Makefile~)

File: batch_job/src/batch_job.h~ [hunk 2]
```diff
@@ -0,0 +1,190 @@
+/*
```
Comment (btovar) [inline batch_job/src/batch_job.h~]: Another backup file (batch_job.h~)

**Comment by dthain** [comment]
I am embarrased!

**Comment by btovar** [comment]
I like a lot having batch_job as its own thing.

**Comment by dthain** [comment]
@btovar I'll leave it to you to merge.

**Comment by btovar** [comment]
Got it.

File: batch_job/src/#batch_job_hadoop.c# [hunk 3]
```diff
@@ -1,3 +1,8 @@
+/*
```
Comment (btovar) [inline batch_job/src/#batch_job_hadoop.c#]: Is the name between #...# on purpose?
If it is not used, why are we keeping the file?

**Comment by btovar** [comment]
Ah! somebody did a commit .

**Comment by nhazekam** [comment]
Sorry! Probably me.

**Pull request merged**